synapse/replication/tcp/commands.py

```diff
@@ -109,14 +115,16 @@ class RdataCommand(Command):
 
     NAME = "RDATA"
 
-    def __init__(self, stream_name, instance_name, token, row):
+    def __init__(
+        self, stream_name: str, instance_name: str, token: Optional[int], row: StreamRow
+    ):
         self.stream_name = stream_name
         self.instance_name = instance_name
         self.token = token
         self.row = row
 
     @classmethod
-    def from_line(cls, line):
+    def from_line(cls: Type["RdataCommand"], line: str) -> "RdataCommand":
```

Comment:
Just for info: [`typing.Self`](https://www.python.org/dev/peps/pep-0673/#use-in-classmethod-signatures) might make this easier in the future. I think it's in new/bleeding edge `typing_extensions` but probably not something we want to opt into for now.

Reply:
Ah, had not seen that before! I was following a pattern we use in a few other spots too.